### PR TITLE
Test for certificate error and prompt to regenerate

### PR DESCRIPTION
**Pull request by brandonweeks**
This resolves an issue where if the `${DOCKER_MACHINE_NAME}` VM was ever removed from `docker-machine` the build process would enter an infinite loop, with the following error being redirected to `/dev/null`.

> Error checking TLS connection: Error checking and/or regenerating the certs: There was an error validating certificates for host "192.168.99.102:2376": open /Users/weeks/.docker/machine/machines/kube-dev/server.pem: no such file or directory You can attempt to regenerate them using 'docker-machine regenerate-certs [name]'. Be advised that this will trigger a Docker daemon restart which will stop running containers.

Now the output is tested for the error string and if found prompts the user if they want to regenerate the certificate. Another option would be setting the `--force` flag on `docker-machine regenerate-certs`, so user intervention wouldn't be required.

/cc @mhbauer

---

This change is [Reviewable](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/24992)

### CODE REVIEW

File: build/common.sh [hunk 1]
```diff
@@ -199,8 +199,13 @@ function kube::build::prepare_docker_machine() {
   }
   docker-machine start "${DOCKER_MACHINE_NAME}" > /dev/null
   # it takes `docker-machine env` a few seconds to work if the machine was just started
-  while ! docker-machine env ${DOCKER_MACHINE_NAME} &> /dev/null; do
-    sleep 1
+  while ! docker_machine_out=$(docker-machine env ${DOCKER_MACHINE_NAME} 2>&1); do
+    if [[ ${docker_machine_out} == *"Error checking TLS connection"* ]]; then
+      echo ${docker_machine_out}
+      docker-machine regenerate-certs ${DOCKER_MACHINE_NAME}
```
Comment (MHBauer) [inline build/common.sh]: `docker-machine` has a prompt. There is no prompt in the `common.sh` script itself.

**Comment by MHBauer** [comment]
I would only agree with adding `--force` if you added an explicit prompt before running it anyway. So, six of one, half-dozen of the other.

File: build/common.sh [hunk 2]
```diff
@@ -199,8 +199,13 @@ function kube::build::prepare_docker_machine() {
   }
   docker-machine start "${DOCKER_MACHINE_NAME}" > /dev/null
   # it takes `docker-machine env` a few seconds to work if the machine was just started
-  while ! docker-machine env ${DOCKER_MACHINE_NAME} &> /dev/null; do
-    sleep 1
+  while ! docker_machine_out=$(docker-machine env ${DOCKER_MACHINE_NAME} 2>&1); do
```
Comment (mikedanese) [inline build/common.sh]: can you declare this local:
```
local docker_machine_out
while ...
```
so it's scoped to the function.

File: build/common.sh [hunk 3]
```diff
@@ -199,8 +199,13 @@ function kube::build::prepare_docker_machine() {
...
+    if [[ ${docker_machine_out} == *"Error checking TLS connection"* ]]; then
+      echo ${docker_machine_out}
```
Comment (mikedanese) [inline build/common.sh]: can you double quote the variable "${...}"

File: build/common.sh (hunk 1 above)
Comment (mikedanese) [inline build/common.sh]: can you double quote the variable "${...}"

File: build/common.sh [hunk 4]
```diff
@@ -199,8 +199,13 @@ function kube::build::prepare_docker_machine() {
   }
   docker-machine start "${DOCKER_MACHINE_NAME}" > /dev/null
   # it takes `docker-machine env` a few seconds to work if the machine was just started
-  while ! docker-machine env ${DOCKER_MACHINE_NAME} &> /dev/null; do
-    sleep 1
+  while ! docker_machine_out=$(docker-machine env ${DOCKER_MACHINE_NAME} 2>&1); do
+    if [[ ${docker_machine_out} == *"Error checking TLS connection"* ]]; then
```
Comment (mikedanese) [inline build/common.sh]: you could just use =~ instead of the wildcards. e.g.:
```console
$ [[ "hhhiii" =~ "hi" ]] && echo true
true
```

**Comment by brandonweeks** [comment]
@mikedanese changes made and rebased.

**Comment by brandonweeks** [comment]
@mikedanese build flake?

**Comment by mikedanese** [comment]
@k8s-bot e2e test this please, issue: #IGNORE

**Comment by brandonweeks** [comment]
Looks like #27462.

**Pull request merged**